packages/cli/src/util/deploy/create-git-meta.ts

```diff
@@ -0,0 +1,155 @@
+import fs from 'fs-extra';
+import { join } from 'path';
+import ini from 'ini';
+import git from 'git-last-commit';
+import {
+  BitbucketMeta,
+  GitHubMeta,
+  GitLabMeta,
+  GitMeta,
+  RepoData,
+} from '../../types';
+import { Output } from '../output';
+
+function getLastCommit(directory: string): Promise<git.Commit> {
+  return new Promise((resolve, reject) => {
+    git.getLastCommit(
+      (err, commit) => {
+        if (err) return reject(err);
+        resolve(commit);
+      },
+      { dst: directory }
+    );
+  });
+}
+
+export async function getRepoData(configPath: string, output: Output) {
+  let gitConfig;
+  try {
+    gitConfig = ini.parse(await fs.readFile(configPath, 'utf-8'));
+  } catch (error) {
+    output.debug(`Error while parsing repo data: ${error.message}`);
+  }
+  if (!gitConfig) {
+    return;
+  }
+
+  const originUrl = gitConfig['remote "origin"']?.url;
+  if (originUrl) {
+    return parseRepoUrl(originUrl);
+  }
+}
+
+export function parseRepoUrl(originUrl: string): RepoData | null {
+  const isSSH = originUrl.startsWith('git@');
+  // Matches all characters between (// or @) and (.com or .org)
+  const provider = originUrl.match(/(?<=(\/\/|@)).*(?=(.com|.org))/);
+  if (!provider) {
+    return null;
+  }
+
+  let org;
+  let repo;
+
+  if (isSSH) {
+    org = originUrl.split(':')[1].split('/')[0];
+    repo = originUrl.split('/')[1].replace('.git', '');
+  } else {
+    // Assume https:// or git://
+    org = originUrl.split('/')[3];
+    repo = originUrl.split('/')[4].replace('.git', '');
```

Comment:
Are there scenarios when [3] or [4] are null/undefined? These can be handled by checking the length or better yet, check `typeof string`

Reply:
I think this undefined behavior is okay because it's handled by `createGitMeta()`, which doesn't return anything (and therefore the git metadata isn't populated on the dashboard as expected) if the result of `parseRepoUrl()` is null or undefined. I agree though that it's probably better for readability to explicitly handle the error and return null. 👍

Reply:
Add a test like this:

```
const parsedUrl = parseRepoUrl('https://github.com/borked');
```

Then you'll see 

```
TypeError: Cannot read properties of undefined (reading 'replace')
```

Reply:
Ohhh yes you're right